### PR TITLE
Add `critical-section` feature.

### PR DESCRIPTION
**Pull request by reitermarkus**
Depends on https://github.com/rust-embedded/svd2rust/pull/651.

### CODE REVIEW

**Comment by adamgreig** [comment]
Any reason not to add critical-section to the default features here? In fact I wonder if it shouldn't even be optional in svd2rust for cortex-m targets.

I guess the main question is: if you don't use `take()` but still have the feature enabled, and don't provide a critical-section impl, do you get a linker error (because it can't find the impl) or not (because the call is optimised out)? If there's no error in that case then I don't think there's any need to even have it as an optional feature; if there is then at least we might consider making it default since the majority of users are likely to need it and it's unlikely a dependency accidentally enables it (same deal as `rt`).

**Comment by reitermarkus** [comment]
Yeah, I guess it can be a default feature here since most likely you want to use `take` rather than `steal`.

I think it should still stay optional in `svd2rust` though, mainly because it *can* be made optional. The PAC crate can then decide if it should be enabled by default.

**Comment by burrbull** [comment]
0.26 is released

**Comment by reitermarkus** [comment]
@burrbull, thanks, rebased.

@adamgreig, this should be good to go.

File: scripts/tool_install.sh [hunk 1]
```diff
@@ -1,30 +1,20 @@
 #!/usr/bin/env bash
 
+set -euo pipefail
+
 FORM_VERSION="${FORM_VERSION:-v0.10.0}"
 SVDTOOLS_VERSION="${SVDTOOLS_VERSION:-v0.2.6}"
```
Comment (burrbull) [inline scripts/tool_install.sh:6]:
```suggestion
SVDTOOLS_VERSION="${SVDTOOLS_VERSION:-v0.2.7}"
```

File: scripts/makecrates.py [hunk 2]
```diff
@@ -72,21 +71,18 @@
 license = "MIT/Apache-2.0"
 
 [dependencies]
-bare-metal = "1.0.0"
+critical-section = {{ version = "1.0", optional = true }}
+cortex-m = "0.7.6"
+cortex-m-rt = {{ version = ">=0.6.15,<0.8", optional = true }}
```
Comment (burrbull) [inline scripts/makecrates.py:76]: @adamgreig Do we still need this workaround?

**Review by adamgreig — APPROVED** [review]
Rebased onto master to resolve conflicts. Thanks for the script updates as well!

bors r+

File: scripts/makecrates.py (hunk 2 above)
Comment (adamgreig) [reply scripts/makecrates.py:76]: It doesn't hurt, we are still compatible with cortex-m-rt 0.6.15 and lets people still use it if they want.

**Pull request merged**